src/main/java/ch/njol/skript/effects/EffContinue.java

```diff
@@ -52,20 +60,18 @@ public class EffContinue extends Effect {
 	}
 
 	@SuppressWarnings("NotNullFieldNotInitialized")
-	private TriggerSection section;
+	private LoopSection loop;
 
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		List<TriggerSection> currentSections = ParserInstance.get().getCurrentSections().stream()
-			.filter(s -> s instanceof SecLoop || s instanceof SecWhile)
-			.collect(Collectors.toList());
+		List<LoopSection> currentLoops = getParser().getCurrentSections(LoopSection.class);
 		
-		if (currentSections.isEmpty()) {
-			Skript.error("Continue may only be used in while or loops");
+		if (currentLoops.isEmpty()) {
+			Skript.error("'Continue' effect may only be used in while loops and regular loops");
```

Comment:
```suggestion
			Skript.error("The 'continue' effect may only be used in loop sections");
```

Reply:
Sure, I will just try to keep the while loop part due to [this review](https://github.com/SkriptLang/Skript/pull/4839#pullrequestreview-1024361047)